### PR TITLE
Search UI

### PR DESCRIPTION
**Pull request by lihter**
- Deleted old HomePageV1 files
- Transformed MovieCell from UICollectionViewCell to UITableViewCell
- Refactored HomePageTwoVC, now it consists of MovieSearchBar and 2 child VCs
- Created CategoriesVC (and CategoriesPresenter and Delegate) which is 99% the same code from HomePageTwoVC before refactoring
- Created SearchVC (+ presenter + delegate) which consists of tableview that presents movies searched in MovieSearchBar

https://user-images.githubusercontent.com/82365531/129562778-3dd04b5a-6536-4075-838b-0d608c9e2bcd.mp4

### CODE REVIEW

File: MovieApp/MovieApp/App/Coordinator/AppRouter.swift [hunk 1]
```diff
@@ -45,7 +45,12 @@ extension AppRouter {
         let tabBarController = UITabBarController()
         
         let homePagePresenter = HomePageTwoPresenter(useCase: moviesUseCase, router: self)
-        let homePageVC = HomePageTwoViewController(presenter: homePagePresenter)
+        let categoriesPresenter = CategoriesPresenter(useCase: moviesUseCase, router: self)
+        let searchPresenter = SearchPresenter(useCase: moviesUseCase, router: self)
+        let homePageVC = HomePageTwoViewController(
```
Comment (fban1) [inline MovieApp/MovieApp/App/Coordinator/AppRouter.swift]: Could it be named HomePageViewController?

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell+Design.swift [hunk 2]
```diff
@@ -54,24 +50,30 @@ extension MovieCell: DesignProtocol {
...
     
     func defineLayoutForViews() {
-        movieImageView.snp.makeConstraints {
-            $0.leading.equalToSuperview()
+        contentView.snp.makeConstraints {
             $0.top.equalToSuperview()
-            $0.height.equalToSuperview()
+            $0.bottom.equalToSuperview().inset(12)
+            $0.leading.equalToSuperview().offset(18)
+            $0.trailing.equalToSuperview().inset(18)
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell+Design.swift]: leading can be inset and you can make it a one liner

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift [hunk 3]
```diff
@@ -1,34 +1,35 @@
...
 import Kingfisher
 
-class MovieCell: UICollectionViewCell {
+class MovieCell: UITableViewCell {
     
     static let reuseIdentifier = String(describing: MovieCell.self)
     
-    let descriptionFontSize: CGFloat = 14
+    var movieId: Int!
+
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift]: no need for this

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController.swift [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        
+        cell.populate(with: movie)
+        cell.selectionStyle = .none
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let cell = tableView.cellForRow(at: indexPath) as? MovieCell else { return }
+        
+        presenter.selectedMovie(withId: cell.movieId)
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController.swift]: you can get a movie id like this

```swift
let id = movies[indexPath.row].id
```

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift [hunk 5]
```diff
@@ -1,34 +1,35 @@
...
         buildViews()
     }
     
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
     
-    func populate(withMovie movie: MovieViewModel) {
+    func populate(with movie: MovieViewModel) {
+        movieId = movie.identifier
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift]: no need for this. Cell doesn't need to have a variable to save ids. You can can get that from UITableViewDataSource didSelectRowAt

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController+Design.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    
+    func createViews() {
+        tableView = UITableView(frame: .zero, style: .grouped)
+        view.addSubview(tableView)
+    }
+    
+    func styleViews() {
+        view.backgroundColor = .white
+        
+        tableView.rowHeight = 142 + 12
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController+Design.swift]: why don't you write 154?

**Review by fban1 — CHANGES_REQUESTED** [review]

File: MovieApp/MovieApp/App/Coordinator/AppRouter.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Coordinator/AppRouter.swift]: Renamed everywhere.
[Here](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-b8e8fb675a7084dca12ae6abb54b13ee1d7293192f1e1e75308b4dca15ea5465R47)

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell+Design.swift (hunk 2 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell+Design.swift]: [Here](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-4e7b2efbcaaa979e7e77ed095dcb6e1fae870d3f9216aa6bbf6a65457310e589R56)
Also replaced everywhere else where it was possible to apply the same code style fix.

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift (hunk 3 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift]: [Removed](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-59acdf9fea7a34d1960015778fa4fae8eba6ce40a9b91286052e7d63a97de079L8)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController.swift (hunk 4 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController.swift]: [Changed](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-6c7e4d8b35d5ff6def2dc66860ecfdc7d2f474e451cb72934495033eeeab86a1R70)

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift (hunk 5 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift]: [Removed](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-59acdf9fea7a34d1960015778fa4fae8eba6ce40a9b91286052e7d63a97de079L28)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController+Design.swift (hunk 6 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/SearchVC/SearchViewController+Design.swift]: [Here](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-aad6f6394c4bd80772719da5c89e1ef095add811bfc58f7abd070251bcbbeba6R20)

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift (hunk 5 above)
Comment (fban1) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCellHPv1/MovieCell.swift]: [LINK MADAFAKA](https://github.com/lihter/Movie-App/pull/41/commits/3383b62f0e45eb7e1f5ba5b650a01beaa73387a1#diff-6a7e0c3cc088b3a980b8a92c08ad5d190f909d67f14aa13f150bdc992ab52302R145)

**Review by fban1 — APPROVED** [review]

**Pull request merged**